strawberry/federation/schema.py

```diff
@@ -210,10 +209,14 @@ def entities_resolver(
             try:
                 result = get_result()
             except Exception as e:  # noqa: PERF203
-                result = GraphQLError(
-                    f"Unable to resolve reference for {definition.origin}",
-                    original_error=e,
-                )
+                if type(e).__name__ == "TypeError":
+                    # check explicitly for type name instead of `isinstance` so
+                    # clients can raise custom TypeErrors to avoid this wrapper
+                    result = Exception(
+                        f"Unable to resolve reference for {type_.definition.name}"
+                    )
+                else:
```

Comment:
Hi, could you elaborate on why `TypeError` should still be wrapped in a new `Exception`?

Reply:
Yes, probably worth a comment. It's for when the query is made with incorrect data, e.g. the `test_fails_properly_when_wrong_data_is_passed` test
https://github.com/strawberry-graphql/strawberry/pull/3144/files#diff-7e4c37801e2121b1d2d8f4cafddcf32731d47271b9097beeb23808ac9ca42196R174

I wrapped it in the Exception to preserve what I considered to be existing behavior from that test that already existed. I can't remember off the top of my head though what the message would be if it weren't wrapped...

I could look into that if you'd like

Reply:
@DoctorJohn I've updated this a bit, I hope it's a little more clear now.

The failures seem to be around code coverage within my tests so they don't need important?

Reply:
@DoctorJohn bumping one more time, update is in this commit: https://github.com/strawberry-graphql/strawberry/pull/3144/commits/56e89556220fa3b60d8a3bfe5c4cd1bd754f79ca

Reply:
Thank you @bradleyoesch, the only reason this is not merged yet is still this bit here. You mentioned:

> I wrapped it in the Exception to preserve what I considered to be existing behavior from that test that already existed.

I want to take a closer look and see whether we need this bit or whether we can avoid it. Sorry this is taking way to long to land. I was a way for quite a while and now picking up all the unfinished work I left.

Reply:
@DoctorJohn yeah no problem! I've forgotten about this several times myself :)

I would like to avoid it too, or at least have a cleaner solution than the type name checking going on, without breaking existing behavior too much

Reply:
Hey @bradleyoesch, I just spend some time on this and refactor this part, finally bringing this over the finish line. I hope it's okay I pushed to your branch.

Reply:
Love it thank you!!